### PR TITLE
Remove old knative revisions in function-controller reconcile

### PR DESCRIPTION
**Pull request by aerfio**
**Description**

Changes proposed in this pull request:

- as in title
- cleanup a bit, separate a few things to distinct files
- fix generation of rbacs, so that they can be copy-pasted

**Related issue(s)**

See also https://github.com/kyma-project/kyma/issues/7947

### CODE REVIEW

File: components/function-controller/internal/controllers/serverless/reconcile.go [hunk 1]
```diff
@@ -0,0 +1,141 @@
...
+	}
+}
+
+func (r *FunctionReconciler) SetupWithManager(mgr ctrl.Manager) error {
+	return ctrl.NewControllerManagedBy(mgr).
+		For(&serverlessv1alpha1.Function{}).
+		Owns(&corev1.ConfigMap{}).
+		Owns(&batchv1.Job{}).
+		Owns(&servingv1.Service{}).
+		Owns(&servingv1.Revision{}).
```
Comment (aerfio) [inline components/function-controller/internal/controllers/serverless/reconcile.go:75]: this line is the only one I added while moving this functionality to separate file

File: components/function-controller/internal/controllers/serverless/reconcile.go [hunk 2]
```diff
@@ -0,0 +1,142 @@
...
+// Reconcile reads that state of the cluster for a Function object and makes changes based on the state read and what is in the Function.Spec
+// +kubebuilder:rbac:groups="",resources=configmaps;secrets;serviceaccounts;namespaces,verbs=get;list;watch;create;update;patch;delete
+// +kubebuilder:rbac:groups="serverless.kyma-project.io",resources=functions,verbs=get;list;watch;create;update;patch;delete
+// +kubebuilder:rbac:groups="serverless.kyma-project.io",resources=functions/status,verbs=get;update;patch
+// +kubebuilder:rbac:groups="serving.knative.dev",resources=services;revisions,verbs=get;list;watch;create;update;patch;delete;deletecollection
+// +kubebuilder:rbac:groups="serving.knative.dev",resources=services/status,verbs=get
+// +kubebuilder:rbac:groups="batch",resources=jobs,verbs=get;list;watch;create;update;patch;delete;deletecollection
+// +kubebuilder:rbac:groups="batch",resources=jobs/status,verbs=get
+// +kubebuilder:rbac:groups="",resources=events,verbs=create;patch
+
```
Comment (aerfio) [inline components/function-controller/internal/controllers/serverless/reconcile.go:88]: I've made some changes here so that we can generate rbacs again

**Comment by michal-hudy** [comment]
I would rather create a separate controller that will monitor Knative Service. Thanks to that logic for cleaning of Knative Revisions will be separated from handling a Function CR. Removing is not related to Function CR and we don't create them, they are created by Knative under the hood.

**Comment by aerfio** [comment]
Closing in favor of #8077

**Pull request closed without merging**